sources/input_line_eff.ml

```diff
@@ -10,9 +13,20 @@ let rec sum_up acc =
     sum_up acc
 
 let _ =
+  Printf.printf "Starting up. Please input:\n%!";
   let r = ref 0 in
-  try sum_up r with
-  | End_of_file -> Printf.printf "Sum is %d\n" !r
-  | effect (Conversion_failure s) k ->
-      Printf.fprintf stderr "Conversion failure \"%s\"\n%!" s;
-      continue k 0
+  match_with sum_up r
+  { effc = (fun (type a) (e: a Effect.t) ->
+      match e with
+      | Conversion_failure s -> Some (fun (k: (a,_) continuation) ->
+              Printf.fprintf stderr "Conversion failure \"%s\"\n%!" s;
+              continue k 0)
+      | _ -> None
+    );
+    exnc = (function
+        | End_of_file -> Printf.printf "Sum is %d\n" !r
+        | e -> raise e
+    );
+    (* Shouldn't reach here, means sum_up returned a value *)
+    retc = fun _ -> failwith "Impossible?" 
```

Comment:
Yes, control never reaches the `retc`-clause, because the handled computation only terminates when `input_line` throws an exception.